rebasehelper/tests/test_application.py

```diff
@@ -112,16 +128,48 @@ def test_application_sources(self, workdir):
                                      False]}},
             'results_dir': os.path.join(workdir, constants.RESULTS_DIR)}
 
-        cli = CLI(self.cmd_line_args)
-        config = Config()
-        config.merge(cli)
-        execution_dir, results_dir, debug_log_file = Application.setup(config)
-        app = Application(config, execution_dir, results_dir, debug_log_file)
         app.prepare_sources()
         for key, val in app.kwargs.items():
             if key in expected_dict:
                 assert val == expected_dict[key]
 
+    def test_setup(self, config):
+        config = config(self.cmd_line_args)
+        files = Application.setup(config)
+        for file in files:
+            assert os.path.exists(file)
+
+    def test_setup_continue(self, config):
+        config = config(self.cmd_line_args + ['--continue'])
+        files = Application.setup(config)
```

Comment:
I don't like this. You are already testing this in the previous test, so the check below is redunant. I also don't like referring to `results_dir` as `files[1]`.

Reply:
Well, the assertions are the same as in the previous test but the directories are created by different branches of code so it imho makes sense to test it here as well. I agree with your second point, will change that.